class.eventcalendar.plugin.php

```diff
@@ -1,265 +1,279 @@
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
+      $Sender->AddDefinition('EventCalendarCategoryIDs', json_encode(C('Plugins.EventCalendar.CategoryIDs')));
+
+      // initially don't hide elements in allowed categories
+      $CategoryID = $Sender->Discussion->CategoryID;
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs'))) {
+         $Hidden = ' Hidden';   
+      }
+	  
+	  //Removed when switching drop down fields to a date picker
+      //$Year = date('Y');
+      //$YearRange = $Year.'-'.($Year + 1);
+
+      $HtmlOut = <<< EOT
+<div class="P EventCalendarInput{$Hidden}">
+   {$Sender->Form->Label('Event Date', 'EventCalendarDate')}  
+	//Updated when switching drop down fields to a date picker
+   {$Sender->Form->Input('EventCalendarDate', 'EventCalendarDate')}
+</div>
+EOT;
+//{$Sender->Form->Date('EventCalendarDate', array('YearRange' => $YearRange, 'fields' => array('day', 'month', 'year')))} //from above
+
+      echo $HtmlOut;
+   } // End of PostController_BeforeBodyInput_Handler
+
+   /**
+    *  Add Validation for event date and check permissions
+    *
+    * @param object $Sender DiscussionModel
+    */
+   public function DiscussionModel_BeforeSaveDiscussion_Handler($Sender) {
+      $Session = Gdn::Session();
+      $CategoryID = $Sender->EventArguments['FormPostValues']['CategoryID'];
+      // Reset event date and return if wrong category or no right to add event
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs')) || !$Session->CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         $Sender->EventArguments['FormPostValues']['EventCalendarDate'] = '';
+         return;
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);	  
+	  $formatted_date = date('Y-m-d',$time);	  
+	  $Sender->EventArguments['FormPostValues']['EventCalendarDate'] = $formatted_date;  
+
+      // Add custom validation text
+      $Sender->Validation->ApplyRule('EventCalendarDate', 'Required', T('Please enter an event date'));
+      $Sender->Validation->ApplyRule('EventCalendarDate', 'Date', T('The event date you\'ve entered is invalid'.'<pre>'.$Sender->EventArguments['FormPostValues']['EventCalendarDate'].' formatted to '.$formatted_date.'</pre>'));
+   } // End of DiscussionModel_BeforeSaveDiscussion_Handler
+
+   /**
+    * Returns nicely formatted html for an event date 
+    *
+    * @param date $EventDate Date of the event
+    * @return string html string showing the event date (translatable)
+    */ 
+   private function FormatEventCalendarDate($EventDate, $IncludeIcon = FALSE) {
+      if(!CheckPermission(array('Plugins.EventCalendar.View'))) {
+         return;
+      }
+      if ($EventDate != '0000-00-00' & !is_null($EventDate))  {
```

Comment:
I never officially learned PHP and that's why I'm not ashamed to ask: mustn't it be "&&" instead of "&"?
But I would prefer to change the function to `private function FormatEventCalendarDate($EventDate = '0000-00-00' , $IncludeIcon = FALSE) {` to avoid the null check